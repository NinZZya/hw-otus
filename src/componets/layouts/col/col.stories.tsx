import React from "react";
/** @jsx jsx */
import { jsx } from "@emotion/core";
// also exported from "@storybook/react" if you can deal with breaking changes in 6.1
import { Meta } from "@storybook/react/types-6-0";

import { Col } from "./col";

export default {
  title: "Example/layouts/Col",
  component: Col,
} as Meta;

const style = { padding: "1rem", border: "1px dotted #000" };

export const ColWithElements: React.FC = () => (
  <Col style={style}>
    <p css={style}>Item 1</p>
    <p css={style}>Item 2</p>
  </Col>
);
