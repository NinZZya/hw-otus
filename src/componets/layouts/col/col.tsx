import React from "react";
/** @jsx jsx */
import { jsx, css } from "@emotion/react";
import { SerializedStyles } from "@emotion/utils/types";

interface ColProp {
  [key: string]: string | undefined | Element[] | React.ReactNode[];
  justifyContent?: string;
  alignItems?: string;
  margin?: string;
  padding?: string;
}

const DefaultStyle = {
  JUSTIFY_CONTENT: "start",
  ALIGIN_ITEMS: "start",
  MARGIN: "0",
  PADDING: "1rem",
};

const getColStyle = (style: ColProp): SerializedStyles => css`
  display: flex;
  justify-content: ${style.justifyContent
    ? style.justifyContent
    : DefaultStyle.JUSTIFY_CONTENT};
  align-items: ${style.alignItems
    ? style.alignItems
    : DefaultStyle.ALIGIN_ITEMS};
  width: 100%;
  height: 100%;
  margin: ${style.margin ? style.margin : DefaultStyle.MARGIN};
  padding: ${style.padding ? style.padding : DefaultStyle.PADDING};
  box-sizing: border-box;
`;

export const Col: React.FC<ColProp> = (props) => {
  return (
    <div css={getColStyle(props)}>
      {React.Children.map(props.children, (child) => child)}
    </div>
  );
};
