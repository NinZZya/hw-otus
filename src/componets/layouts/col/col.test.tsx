import React from "react";
import renderer from "react-test-renderer";
import { Col } from "./col";

describe(`Col render coorectly`, () => {
  it(`Col without props render coorectly`, () => {
    const tree = renderer.create(<Col />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Col with components render coorectly`, () => {
    const tree = renderer
      .create(
        <Col>
          <div>div</div>
          <div>div</div>
          <div>div</div>
        </Col>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
