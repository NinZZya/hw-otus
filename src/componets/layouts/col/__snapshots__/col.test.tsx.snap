// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col render coorectly Col with components render coorectly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "9te4wq",
      "next": undefined,
      "styles": "
  display: flex;
  justify-content: start;
  align-items: start;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 1rem;
  box-sizing: border-box;
",
      "toString": [Function],
    }
  }
>
  <div>
    div
  </div>
  <div>
    div
  </div>
  <div>
    div
  </div>
</div>
`;

exports[`Col render coorectly Col without props render coorectly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "9te4wq",
      "next": undefined,
      "styles": "
  display: flex;
  justify-content: start;
  align-items: start;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 1rem;
  box-sizing: border-box;
",
      "toString": [Function],
    }
  }
/>
`;
